lib/node_modules/@stdlib/lapack/base/dlacpy/docs/types/index.d.ts

```diff
@@ -0,0 +1,118 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.1
+
+/// <reference types="@stdlib/types"/>
+
+import { Layout } from '@stdlib/types/blas';
+
+/**
+* Interface describing `dlacpy`.
+*/
+interface Routine {
+	/**
+	* Copies all or part of a two-dimensional matrix `A` to another matrix `B`.
+	*
+	* @param order - storage layout of `A` and `B`
+	* @param uplo - specifies whether the upper or lower triangular part of matrix `A` is supplied
+	* @param M - number of rows in matrix `A`
+	* @param N - number of columns in matrix `A`
+	* @param A - input matrix
+	* @param LDA - stride of the first dimension of `A` (a.k.a., leading dimension of the matrix `A`)
+	* @param B - destination matrix
+	* @param LDB - stride of the first dimension of `B` (a.k.a., leading dimension of the matrix `B`)
+	* @returns `B`
+	*
+	* @example
+	* var Float64Array = require( '@stdlib/array/float64' );
+	*
+	* var A = new Float64Array( [ 1.0, 2.0, 3.0, 4.0 ] );
+	* var B = new Float64Array( 4 );
+	*
+	* dlacpy( 'row-major', 'all', 2, 2, A, 2, B, 2 );
+	* // B => <Float64Array>[ 1.0, 2.0, 3.0, 4.0 ]
+	*/
+	( order: Layout, uplo: string, M: number, N: number, A: Float64Array, LDA: number, B: Float64Array, LDB: number ): Float64Array;
```

Comment:
Either we should be stricter with `uplo` (e.g., require that the only acceptable value other than `upper` or `lower` is `all`), or we should at least do `uplo: UPLO | string`, so that users can a better type hint than "string" in their IDEs.